internal/pkg/bulk/engine.go

```diff
@@ -336,6 +336,9 @@ func (b *Bulker) Run(ctx context.Context) error {
 
 	w := semaphore.NewWeighted(int64(b.opts.maxPending))
 
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
```

Comment:
This causes each call to the bulker's Run() function to return unconditionally after 5 minutes, including the one in the fleet server main loop.

You can see this happening in the logs. Every 5 minutes the bulker aborts everything it is doing.

To avoid this unconditional exit you would create the context in doFlush https://github.com/juliaElastic/fleet-server/blob/a9a82eaae87b7c2d4b54d714d34943a970c8f3cd/internal/pkg/bulk/engine.go#L352

That would bound each execution of `doFlush` to 5 minutes which should have the same effect, but without the logging about constant exiting.

Reply:
I tried moving the timeout to `doFlush` but it doesn't work, I think the `doFlush` function exits too quickly and cancels the context.
See the logs from a 10k run: https://platform-logging.kb.us-west2.gcp.elastic-cloud.com/app/r/s/S0Wla

Reply:
I moved the timeout outside of `doFlush`, see this comment: https://github.com/elastic/ingest-dev/issues/3783#issuecomment-2459697129
Strangely I don't really see the flush timing out anymore, mostly seeing the upgrade step being stuck at 75k runs, 10k runs pass almost always.